packages/web/src/pages/search-page-v2/SearchFilters.tsx

```diff
@@ -109,76 +109,56 @@ const KeyFilter = () => {
   const [scale, setScale] = useState(key?.split(' ')[1] ?? 'Major')
   const keyOptions = MUSICAL_KEYS.map((k) => ({
     label: k,
-    value: getValueFromKey(k)
+    value: `${getValueFromKey(k)} ${scale}`
   }))
 
-  const activeValue = useMemo(() => {
-    if (!key) return null
-    const keyRegex = /(.+) (Major|Minor)/
-    const keyRes = keyRegex.exec(key)?.[1]
-    return keyRes ? getValueFromKey(keyRes) : null
-  }, [key])
-
-  const label = useMemo(() => {
-    const pitch = key?.split(' ')[0]
-    return keyOptions.find((option) => option.value === pitch)?.label
-  }, [key, keyOptions])
+  const renderLabel = useCallback(
+    (label: string) => {
+      const musicalKey = MUSICAL_KEYS.find((key) => key.match(label))
+      if (!musicalKey) return messages.key
+      return `${musicalKey} ${scale}`
+    },
+    [scale]
+  )
 
   return (
     <FilterButton
       value={key}
-      label={label ? `${label} ${scale}` : messages.key}
+      renderLabel={renderLabel}
+      label={messages.key}
       onChange={updateSearchParams}
-      iconRight={IconCaretDown}
+      popupProps={{
+        anchorOrigin: { vertical: 'bottom', horizontal: 'left' },
+        transformOrigin: { vertical: 'top', horizontal: 'left' },
+        css: { minWidth: 200 }
+      }}
+      options={keyOptions}
     >
-      {({ handleChange, isOpen, setIsOpen, anchorRef }) => (
-        <Popup
-          anchorRef={anchorRef}
-          isVisible={isOpen}
-          onClose={() => setIsOpen(false)}
-          anchorOrigin={{ vertical: 'bottom', horizontal: 'left' }}
-          transformOrigin={{ vertical: 'top', horizontal: 'left' }}
+      {({ options }) => (
+        <Flex
+          w='100%'
+          gap='s'
+          pv='s'
+          direction='column'
+          alignItems='flex-start'
+          role='listbox'
         >
-          <Paper
-            mt='s'
-            border='strong'
-            shadow='far'
-            css={{ minWidth: 200 }}
-            onClick={(e) => e.stopPropagation()}
-          >
-            <Flex
-              w='100%'
-              gap='s'
-              pv='s'
-              direction='column'
-              alignItems='flex-start'
-              role='listbox'
-            >
-              <Box w='100%' ph='s'>
-                <SegmentedControl
-                  fullWidth
-                  options={[
-                    { key: 'Major', text: 'Major' },
-                    { key: 'Minor', text: 'Minor' }
-                  ]}
-                  selected={scale}
-                  onSelectOption={setScale}
-                />
-              </Box>
-              <Divider css={{ width: '100%' }} />
-              <Flex direction='column' w='100%' ph='s'>
-                <FilterButtonOptions
-                  activeValue={activeValue}
```

Comment:
activeValue should work by default now right? This was added so a value coming from the url is highlighted in the options when the filter popup is opened

Reply:
yeah let me confirm if that's needed, i think it should just worked based on "value" now